pkg/cache/context.go

```diff
@@ -632,6 +620,59 @@ func (ctx *Context) PublishEvents(eventRecords []*si.EventRecord) {
 	}
 }
 
+// this function handles the pod scheduling failures with respect to the different causes,
+// and update the pod condition accordingly. the cluster autoscaler depends on the certain
+// pod condition in order to trigger auto-scaling.
+func (ctx *Context) HandleContainerStateUpdate(request *si.UpdateContainerSchedulingStateRequest) {
+	updatePodConditionFn := func(task *Task, podConditionReason, podConditionMsg, eventMessage string) {
```

Comment:
Can we move this big function out of the handle function into its own function?
Make a new function `ctx.updatePodCondition()` with the same parameters.